packages/kit/src/runtime/client/client.js

```diff
@@ -855,9 +855,25 @@ async function load_route({ id, invalidating, url, params, route }) {
 		try {
 			server_data = await load_data(url, invalid_server_nodes);
 		} catch (error) {
+			const handled_error = await handle_error(error, { url, params, route: { id } });
+
+			if (preload) {
```

Comment:
What does the return statement do? I'm wondering what happens if there's an error, and then the user actually clicks on the link? The return statement looks a bit like a placeholder to me.

Reply:
I believe it prevents the load data from being cached. I was trying to match the shape to make the conditional below succeed. https://github.com/sveltejs/kit/blob/6107466adf63cdf43044702f5360fcb497ba81f0/packages/kit/src/runtime/client/client.js#L380-L386
If the user clicks on the link it would try loading again (and trigger the error page if the network is still offline).

Reply:
Maybe something similar like we do today already with the navigation token:
- set a `preload` token (empty object, unique key basically)
- when the user actually clicks on a link, the preload token becomes the navigation token
- when an error occurs, the preload token passed to the navigation is compared with the current navigation token, and if they're the same we know that we can proceed with the logic as-is (i.e. the preload became the real navigation). Else we return a fake error state, or we throw and the preload catches that